tensorflow/lite/micro/kernels/resize_bilinear.cc

```diff
@@ -0,0 +1,112 @@
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/kernels/op_macros.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/micro_utils.h"
+
+namespace tflite {
+namespace {
+
+constexpr int kInputTensor = 0;
+constexpr int kSizeTensor = 1;
+constexpr int kOutputTensor = 0;
+
+TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
+  TF_LITE_ENSURE_EQ(context, NumInputs(node), 2);
+  TF_LITE_ENSURE_EQ(context, NumOutputs(node), 1);
+
+  const TfLiteTensor* input = GetInput(context, node, kInputTensor);
+  const TfLiteTensor* size = GetInput(context, node, kSizeTensor);
+  TfLiteTensor* output = GetOutput(context, node, kOutputTensor);
+
+  TF_LITE_ENSURE_EQ(context, NumDimensions(input), 4);
+  TF_LITE_ENSURE_EQ(context, NumDimensions(size), 1);
+
+  TF_LITE_ENSURE_EQ(context, size->type, kTfLiteInt32);
+  output->type = input->type;
+
+  TF_LITE_ENSURE_MSG(context, IsConstantTensor(size),
+                     "Non constant size tensor not supported");
+
+  // Ensure params are valid.
+  auto* params =
+      reinterpret_cast<TfLiteResizeBilinearParams*>(node->builtin_data);
+  if (params->half_pixel_centers && params->align_corners) {
+    TF_LITE_KERNEL_LOG(
+        context, "If half_pixel_centers is True, align_corners must be False.");
+    return kTfLiteError;
+  }
+
+  return kTfLiteOk;
+}
+
+TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
+  auto* params =
+      reinterpret_cast<TfLiteResizeBilinearParams*>(node->builtin_data);
+
+  const TfLiteEvalTensor* input =
+      tflite::micro::GetEvalInput(context, node, kInputTensor);
+  const TfLiteEvalTensor* size =
+      tflite::micro::GetEvalInput(context, node, kSizeTensor);
+  TfLiteEvalTensor* output =
+      tflite::micro::GetEvalOutput(context, node, kOutputTensor);
+
+  if (output->type == kTfLiteFloat32) {
+#define TF_LITE_RESIZE_BILINEAR(opname, datatype)                        \
+  tflite::ResizeBilinearParams op_params;                                \
+  op_params.align_corners = params->align_corners;                       \
+  op_params.half_pixel_centers = params->half_pixel_centers;             \
+  reference_ops::opname(op_params, tflite::micro::GetTensorShape(input), \
+                        tflite::micro::GetTensorData<datatype>(input),   \
+                        tflite::micro::GetTensorShape(size),             \
+                        tflite::micro::GetTensorData<int32_t>(size),     \
+                        tflite::micro::GetTensorShape(output),           \
+                        tflite::micro::GetTensorData<datatype>(output))
+
+    TF_LITE_RESIZE_BILINEAR(ResizeBilinear, float);
+  } else if (output->type == kTfLiteInt8) {
+    TF_LITE_RESIZE_BILINEAR(ResizeBilinearInteger, int8_t);
+  } else if (output->type == kTfLiteInt16) {
+    TF_LITE_RESIZE_BILINEAR(ResizeBilinearInteger, int16_t);
```

Comment:
let's drop int16 support. This can be added in later, as needed. First op port should have only int8 and float support.

Reply:
fixed